src/cyndilib/video_frame.pyx

```diff
@@ -202,6 +202,14 @@ cdef class VideoFrame:
         cdef bytes result = self.ptr.p_metadata
         return result
 
+    def set_metadata(self, bytes metadata):
+        """Set the video metadata. The lifetime of the metadata must be as long as the VideoFrame.
+        """
+        self._set_metadata(metadata)
+    cdef int _set_metadata(self, bytes metadata) noexcept:
+        self.ptr.p_metadata = metadata
```

Comment:
Since the bytes object is owned by the caller, it might be safer to assign it as an instance attribute of `VideoFrame`.

That way its lifetime will be tied to the underlying frame pointer and handled within `__dealloc__`.

Maybe add something like `cdef readonly bytes _metadata_bytes` to the pxd?  Then:

```cython
cdef int _set_metadata(self, bytes metadata) except -1:
    self._metadata_bytes = metadata
    self.ptr.p_metadata = self._metadata_bytes
```

Then the sentence about lifetime from the `set_metadata` docstring wouldn't be needed